Sean---L203.water.mapping

swd-turner opened the pull request:
#622

swd-turner commented:
hmm... not sure why these checks failed. Was passing all tests on my machine. Will look at this again shortly.

bpbond commented on R/zchunk_L203.water.mapping.R:
```diff
@@ -38,103 +37,134 @@ module_water_L203.water.mapping_DISABLED <- function(command, ...) {
+
+    # Sector information
+    L203.mapping_all %>%
+      select(one_of(c(LEVEL2_DATA_NAMES[["Supplysector"]], "logit.type"))) ->
```
I will add XML-generating file to this PR.

bpbond commented:
Nuts I just screwed this up. @pralitp can you fix `zchunk_batch_water_mapping.xml.R`? Sorry.

pralitp commented:
My first guess is you just forgot the first parameter to `add_logit_tables_xml(L203.SubsectorLogit, "SubsectorLogit") %>%`  but I'm running it now to make sure.

bpbond commented:
Oh yeah sorry about that. :doh:

pralitp commented:
Ok, from an eyeball test the xml looks good.  Unfortunately by XML diff tools are too rudimentary to tell for sure it it exactly matches the old data system.  I wouldn't hold up this PR on those grounds as we will do the XML checks in a final sweep anyways.

skim301 commented on R/zchunk_L203.water.mapping.R:
```diff
@@ -38,103 +37,134 @@ module_water_L203.water.mapping_DISABLED <- function(command, ...) {
+      arrange(GCAM_region_ID) %>%
+      left_join(select(L165.ag_IrrEff_R, -field.eff), by = "GCAM_region_ID") %>%
+      # ^^ non-restrictive join required (NA values generated for region 30, Taiwan)
+      mutate(coefficient = if_else(water.sector == IRRIGATION & water_type == "water withdrawals",
```
Please put in a comment that the conveyance loss is taken out of agricultural withdrawals and placed in as a loss in the water distribution sectors (water_td_irr). Thus, to get comparable total agriculture withdrawals for reporting, the conveyance loss must be included.  This is a database query issue, but I don't where else we can document it.  Also originally, I had used efficiency (conveyance.eff) directly rather than 1/eff, which makes it easier to interpret the conveyance loss since its a fraction less than 1, rather than a coefficient greater than 1. Please put in a comment that maybe we should reconsider that.

swd-turner replied:
Ok - comment added

skim301 commented on R/zchunk_L203.water.mapping.R:
```diff
@@ -38,103 +37,134 @@ module_water_L203.water.mapping_DISABLED <- function(command, ...) {
+
+
+    # OUTPUTS
+    L203.Supplysector %>%
```
I would consider creating the water mapping files by demand types (irr, dom, ind, ...) rather than by variable type and break up the final water mapping xml file by demand types to reduce the file size and allow ease of modification. But this is for future work too.

bpbond replied:
Created #663 to follow up on this.

swd-turner commented:
@bpbond Just made a quick change, updated to master on my windows machine and synced with repo... it seems that all of the files that @rplzzz fixed recently (bad line endings #657 ) automatically reverted back to windows line endings on my branch after I updated from master locally (so don't merge it!). The same type of thing seems to have happened in some of the other pull requests that are active at the moment. I'm thinking windows users have been committing bad files inadvertently as they've updated from master on git desktop (i.e., changes made elsewhere are being converted to windows line endings automatically when they're merged into a branch on windows). Does that sound plausible?

bpbond commented:
Ugh. Yeah, that does sound plausible. Probably we're going to have to live with issue until version 1, and then (i) ask everyone to delete the repo from their local machines, (ii) run git bfg, line-fixing utility, and various other fixes, and (iii) have everyone re-clone.

We could also ask folks to [change their line endings settings](https://stackoverflow.com/questions/10418975/how-to-change-line-ending-settings). Thoughts @rplzzz ?

swd-turner commented:
I think the problem might go away if windows users update from master within the repo rather than in Git Desktop.

swd-turner commented:
... but yes, the line ending settings is what I needed. Can you suggest a clean way to reverse/remove the prior two commits here and start afresh on this branch from an earlier commit? Or do I need to manually fix the files on my branch and then recommit?

bpbond commented:
@swd-turner See the first entry at http://ohshitgit.com.

bpbond commented:
@swd-turner So I think if you fix the line ending issue, and update to latest `master`, this is ready to merge?

swd-turner commented:
Correct. I'll fix this today.

bpbond commented on R/module-helpers.R:
```diff
@@ -353,133 +353,3 @@ get_ssp_regions <- function(pcGDP, reg_names, income_group,
 
   regions$region
 }
-
```
Whoah, why are you removing `fill_exp_decay_extrapolate`?

swd-turner replied:
Ok, problems here with the reset!...

bpbond replied:
I sympathize...but @pralitp will be pissed if you rip out all his recent work 😆

swd-turner replied:
s**t, what a mess!!... you around today?

bpbond replied:
No, sorry! I'm in Washington (state). Hang on let me look.

bpbond replied:
Ugh. Question: what about creating a new branch from `master`, dropping in all your code (no downstream work, right, so 'just' the main chunk and its xml partner, along with potentially some inputs?), and creating a new PR?

I'm concerned that fixing this is going to be beyond my git-fu.

swd-turner replied:
I was thinking the exact same thing!... Well beyond my git-fu. I'll write the new branch and PR first, then we can delete this horrible thing.
